syntax = "proto3";

package mapreduce;


service Worker {
    rpc Reduce(ReduceTask) returns (ReduceResponse) {}
    rpc Map(MapTask) returns (MapResponse) {}
}

message ReduceTask {
    repeated string partition_paths = 1;
    string output_path = 2;
}

message ReduceResponse {}

message MapTask {
    string file_path = 1;
    int32 num_partitions = 2;
    string output_dir = 3;
}

message MapResponse {
    repeated string partition_paths = 1;
}
